### PR TITLE
Update keras_nlp.metrics for core

### PR DESCRIPTION
**Pull request by mattdangerw**
This PR isn't really a port for jax/torch. It is unclear how/if these should work without string dyptes, so just punting for now. But it does two things

- Disables these metrics symbols outside the tf runtime.
- Updates testing so test pass in keras-core tf (keras-core does not allow a evaluate call without a loss, so we switch to compute_metrics)

### CODE REVIEW

File: keras_nlp/metrics/edit_distance_test.py [hunk 1]
```diff
@@ -107,34 +107,30 @@ def test_rank_2_tensor_input_normalize_false(self):
 
     def test_model_compile_normalize(self):
         inputs = keras.Input(shape=(None,), dtype="string")
-        outputs = tf.strings.lower(inputs)
```
Comment (jbischof) [inline keras_nlp/metrics/edit_distance_test.py:110]: Isn't it OK to keep TF code if we assert TF backend?

File: keras_nlp/metrics/bleu_test.py [hunk 2]
```diff
@@ -142,12 +144,10 @@ def test_different_order(self):
 
     def test_model_compile(self):
         inputs = keras.Input(shape=(), dtype="string")
-        outputs = tf.identity(inputs)
-        model = keras.Model(inputs, outputs)
-
+        model = keras.Model(inputs, inputs)
```
Comment (ianstenbit) [inline keras_nlp/metrics/bleu_test.py:147]: IMO it's kinda nice to have the explicit version where we have different outputs, even if we just use `ops.identity`.

Totally an optional nit

**Review by ianstenbit — APPROVED** [review]
🐸

**Review by jbischof — APPROVED** [review]
Thanks!

File: keras_nlp/metrics/edit_distance_test.py (hunk 1 above)
Comment (mattdangerw) [reply keras_nlp/metrics/edit_distance_test.py:110]: It is not. In keras-core, a symbolic input like `keras.Input` cannot be called on tf ops. You would need to wrap this with a layer. Also, none of the tests really seemed to be doing anything with the lower, it was just a holdover.

File: keras_nlp/metrics/bleu_test.py (hunk 2 above)
Comment (mattdangerw) [reply keras_nlp/metrics/bleu_test.py:147]: Similar answer as below. `ops.identity` will fail on `tf.keras`'s `KerasTensor` (they aren't designed to interop), and `tf.identity` will fail on `keras_core`'s `KerasTensor` ([see these lines](https://github.com/keras-team/keras-core/blob/ce9b511a063f0be3988ba34accc2e07dc1812ab9/keras_core/backend/common/keras_tensor.py#L87-L108)).

But I just realized we can use `keras.layers.Identity` in all cases, will make that change.

File: keras_nlp/metrics/edit_distance_test.py (hunk 1 above)
Comment (mattdangerw) [reply keras_nlp/metrics/edit_distance_test.py:110]: Here's [the error](https://github.com/keras-team/keras-core/blob/ce9b511a063f0be3988ba34accc2e07dc1812ab9/keras_core/backend/common/keras_tensor.py#L87-L108) you would get in `keras_core` tf backend btw.

This will be an interesting "gotcha" to see how many people hit when moving to keras-core, but I think that's a question that is out of scope for this PR.
